packages/x-date-pickers/src/managers/useTimeManager.ts

```diff
@@ -35,11 +35,13 @@ export function useTimeManager<TEnableAccessibleFieldDOMStructure extends boolea
         ...getTimeFieldInternalPropsDefaults({ utils, internalProps }),
       }),
       internal_getOpenPickerButtonAriaLabel: ({ value, utils, localeText }) => {
-        const formattedValue = utils.isValid(value) ? utils.format(value, 'fullTime') : null;
+        const formattedValue = utils.isValid(value)
```

Comment:
How about we don't change the behavior for now here?
We should be able to use `utils.is12HourCycleInCurrentLocale()`

On `master`, `internal_getOpenPickerButtonAriaLabel` always respects the current locale, people can't override that.
WDYT to keep that and add a comment that we could add the customization feature later on?

Reply:
Nice catch. 👍 
Honestly, to me, the current behavior seems like a bug. 🤔 
If a 24-hour clock has been requested by the application, maybe that's what the app should display everywhere? 🤔 
WDYT?

Reply:
If you want to do it in the same PR then I would at least update the title.
I agree that the new behavior is probably more coherent.

Reply:
I have extracted the `aria-label` meridiem fix to a separate branch/PR that I'll open on this repo after we merge this one. 😉 
https://github.com/LukasTy/mui-x/pull/3